### DISCUSSION TITLE
webhooks

### DISCUSSION THREAD

**Discussion by jcoffi** [docker/extension-ideas#147] [Ideas]
a simple extension than can send and receive web hooks from docker hub

**Comment by ivan-curkovic** [comment]
Hi @jcoffi,
Can you give an example? What is the use case?

**Reply by jcoffi** [reply]
For example. 

I'm a developer and I have containers running locally. A teammate has updated the image our team is using. I'd like to have the option to know the new image is available.  I'd like to not have to run something like watchtower.

Github repo is updated -> autobuild builds -> autotest tests -> new image for the team is available

Right now that's where it stops. But the ability to continue on that workflow so that users with the extension can choose to be notified, automatically download the latest version or automatically have their image updated (ala watchtower)

**Reply by mikeparker** [reply]
@jcoffi we are thinking about this problem in fact! Out of curiosity, could you give any more details of what this image is doing?

I'd also be curious if you would want to choose which images / tags to be notified / auto update for, or just be able to browse your local image list for available updates.

Another similar feature would be to use the `--pull` flag on `docker build` or `docker compose build` or simply use `docker compose pull`, this would of course only update when you choose to restart your containers, rather than whilst they are already running. https://stackoverflow.com/a/39127792/1554346

**Reply by slimslenderslacks** [reply]
When you say that the "github repo is updated", do you imagine this being an automatic pull request so that your Ci build/tests run automatically?
